app/models/subnet.rb

```diff
@@ -121,7 +121,11 @@ class Jail < ::Safemode::Jail
   def self.network_reorder(order_string = 'network')
     adapter = connection.adapter_name.downcase
     if adapter.starts_with?('postgresql')
-      reorder(order_string.sub('network', 'inet(network)'))
+      if order_string =~ /ASC$/
+        reorder(Arel::Nodes::NamedFunction.new('INET', [Subnet.arel_table[:network].asc]))
+      else
+        reorder(Arel::Nodes::NamedFunction.new('INET', [Subnet.arel_table[:network].desc]))
+      end
```

Comment:
Actually, this would be better

```suggestion
      order_by = case order_string
      when 'network ASC':
        Subnet.arel_table[:network].asc
      when 'network DESC':
        Subnet.arel_table[:network].desc
      else
        Subnet.arel_table[:network]
      end
      
      reorder(Arel::Nodes::NamedFunction.new('INET', [order_by]))
```

Untested, as always. ;)

Reply:
Ok, tests fail with this.
The `.asc` and `.desc` have to be called outside of `INET()`.

This should work:

```suggestion
      subnet_table = Subnet.arel_table
      order_by = Arel::Nodes::NamedFunction.new('INET', [subnet_table[:network]])
      
      reorder = case order_string
      when 'network ASC':
        Arel::Nodes::Ascending.new(order_by)
      when 'network DESC':
        Arel::Nodes::Descending.new(order_by)
      else
        order_by
      end
      
      reorder(reorder)
```